tests/integration/conftest.py

```diff
@@ -14,38 +17,59 @@ def table_name_prefix() -> str:
 
 
 @pytest.fixture
-def endpoint():
+def endpoint() -> Optional[URL]:
     if os.environ.get("TEST_ON_AWS", "false") == "true":
         return None
     if "DYNAMODB_URL" not in os.environ:
         raise pytest.skip("DYNAMODB_URL not defined in environment")
-    return os.environ["DYNAMODB_URL"]
+    return URL(os.environ["DYNAMODB_URL"])
 
 
 @pytest.fixture
-def region():
+def region() -> str:
     return os.environ.get("DYNAMODB_REGION", "us-east-1")
 
 
 @pytest.fixture
-async def client(http, endpoint, region):
+def client(http: HTTP, endpoint: URL, region: str) -> Client:
     yield Client(
-        http,
-        Credentials.auto(),
-        region,
-        URL(endpoint) if endpoint is not None else endpoint,
+        http, Credentials.auto(), region, endpoint,
     )
 
 
 @pytest.fixture
-async def table(client: Client, table_name_prefix: str):
-    name = table_name_prefix + str(uuid.uuid4())
-    await client.create_table(
-        name,
-        Throughput(5, 5),
-        KeySchema(KeySpec("h", KeyType.string), KeySpec("r", KeyType.string)),
-        wait_for_active=WaitConfig(max_attempts=25, retry_delay=5),
-    )
+async def table_factory(
+    client: Client, table_name_prefix: str
+) -> Callable[[Optional[Throughput]], Awaitable[str]]:
+    async def factory(throughput: Throughput = Throughput(5, 5)) -> str:
+        name = table_name_prefix + str(uuid.uuid4())
+        await client.create_table(
+            name,
+            throughput,
+            KeySchema(KeySpec("h", KeyType.string), KeySpec("r", KeyType.string)),
+            wait_for_active=WaitConfig(max_attempts=25, retry_delay=5),
+        )
+        return name
+
+    return factory
+
+
+@pytest.fixture
+async def table(
+    client: Client, table_factory: Callable[[Optional[Throughput]], Awaitable[str]]
+) -> str:
+    name = await table_factory()
+    try:
+        yield name
+    finally:
+        await client.delete_table(name)
```

Comment:
I wonder if it's a good cause for walrus:
```py
try:
    yield (name := await table_factory())
finally:
   await delete(name)
```
---
Though, yeah, technically, it's not quite correct, in case table creation fails 🤔

Reply:
I think the walrus should be used sparingly because it's quite weird. I don't find your version easier to read.